src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.MoveTypeCodeAction.cs

```diff
@@ -54,7 +54,6 @@ protected override async Task<IEnumerable<CodeActionOperation>> ComputeOperation
             internal override bool IsApplicable(Workspace workspace)
                 => _operationKind switch
                 {
-                    MoveTypeOperationKind.RenameFile => workspace.CanRenameFilesDuringCodeActions(_state.SemanticDocument.Document.Project),
                     MoveTypeOperationKind.MoveTypeNamespaceScope => _state.TypeNode.Parent is TNamespaceDeclarationSyntax,
```

Comment:
this is bizarre.  THe purpose of IsApplicable is to do checks that can only be done on hte UI thread.  it's one final check to make sure we can proceed.

The check of `_state.TypeNode.Parent is TNamespaceDeclarationSyntax` can be done when the analyzer runs trivially.  We should do this in the analyzer and remove IsApplicable entirely.